Packages/com.unity.inputsystem/InputSystem/InputSystem.cs

```diff
@@ -2401,7 +2401,9 @@ public static InputEventListener onEvent
         /// <seealso cref="ButtonControl.isPressed"/>
         /// <seealso cref="onEvent"/>
         public static IObservable<InputControl> onAnyButtonPress =>
-            onEvent.Select(e => e.GetFirstButtonPressOrNull()).Where(c => c != null);
+            onEvent
+                .Where(evt => evt.type == StateEvent.Type || evt.type == DeltaStateEvent.Type)
+                .Select(e => e.GetFirstButtonPressOrNull()).Where(c => c != null);
```

Comment:
Wondering, is it better to solve this here for `onAnyButonPress` or should `GetFirstButtonPressOrNull` simply return null for any event that isn't a state/delta event?

Reply:
Thinking about it, think I'd prefer the latter and just allow `GetFirstButtonPressOrNull` to be called on any event.

Reply:
Yep, that's much better

Reply:
Nice!

Just saw that the lazy butt who added the method (bet the history shows it was me) didn't properly document it. Would you mind adding the docs for `eventPtr`, `magnitude`, and `buttonControlsOnly`?

Reply:
Haha, no probs